connectors/sources/google_cloud_storage.py

```diff
@@ -199,17 +148,114 @@ async def _api_call(
                 )
                 await asyncio.sleep(DEFAULT_WAIT_MULTIPLIER**retry_counter)
 
+
+class GoogleCloudStorageDataSource(BaseDataSource):
+    """Google Cloud Storage"""
+
+    name = "Google Cloud Storage"
+    service_type = "google_cloud_storage"
+
+    def __init__(self, configuration):
+        """Set up the connection to the Google Cloud Storage Client.
+
+        Args:
+            configuration (DataSourceConfiguration): Object of DataSourceConfiguration class.
+        """
+        super().__init__(configuration=configuration)
+        self.enable_content_extraction = self.configuration["enable_content_extraction"]
+
+        self.google_storage_client = None
+
+    @classmethod
+    def get_default_configuration(cls):
+        """Get the default configuration for Google Cloud Storage.
+
+        Returns:
+            dictionary: Default configuration.
+        """
+        default_credentials = {
+            "type": "service_account",
+            "project_id": "dummy_project_id",
+            "private_key_id": "abc",
+            "private_key": open(DEFAULT_PEM_FILE).read(),
+            "client_email": "123-abc@developer.gserviceaccount.com",
+            "client_id": "123-abc.apps.googleusercontent.com",
+            "auth_uri": "https://accounts.google.com/o/oauth2/auth",
+            "token_uri": "http://localhost:443/token",
+        }
+        return {
+            "service_account_credentials": {
+                "value": json.dumps(default_credentials),
+                "label": "Google Cloud service account json",
+                "type": "str",
+            },
+            "retry_count": {
+                "value": DEFAULT_RETRY_COUNT,
+                "label": "Maximum retries for failed requests",
+                "type": "int",
+            },
+            "enable_content_extraction": {
+                "value": DEFAULT_CONTENT_EXTRACTION,
+                "label": "Enable content extraction (true/false)",
+                "type": "bool",
+            },
+        }
+
+    async def validate_config(self):
+        """Validates whether user inputs are valid or not for configuration field.
+
+        Raises:
+            Exception: The service account json is empty.
+            Exception: The format of service account json is invalid.
+        """
+        if (
+            self.configuration["service_account_credentials"] == ""
+            or self.configuration["service_account_credentials"] is None
+        ):
+            raise Exception("Google Cloud service account json can't be empty.")
+        try:
+            json.loads(self.configuration["service_account_credentials"])
+        except ValueError:
+            raise Exception("Google Cloud service account is not a valid JSON.")
+
+    def get_storage_client(self):
```

Comment:
Since it's not an async client, you can make it a cached property instead and it will be much cleaner: https://docs.python.org/3/library/functools.html#functools.cached_property